### PR TITLE
CONTRIBUTING

### PR DESCRIPTION
**Pull request by ngama75**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Contributing to SPQlios-fft
+
+The spqlios-fft team encourages contributions.
+We encourage users to fix bugs, improve the documentation, write tests and to enhance the code, or ask for new features.
+We encourage researchers to contribute implementations of their FFT or NTT algorithms.
```
Comment (sguaschsbt) [inline CONTRIBUTING.md]: contribute 'with' implementations?¿

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+All communications are public, so please make sure to maintain professional behaviour in
+all published comments. See [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) for
+guidelines.
+
+## Reporting Bugs or Requesting features ##
+
+Bug should be filed at [https://github.com/tfhe/spqlios-fft/issues](https://github.com/tfhe/spqlios-fft/issues).
+
+Features can also be requested there, in this case, please ensure that the feature you request are self-contained,
```
Comment (sguaschsbt) [inline CONTRIBUTING.md]: feature --> features

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+This section applies to people that already have write access to the repository. Specific instructions for pull-requests
+from public forks will be given later.
+
+To implement some changes, please follow these steps:
+
+- Create a "topic branch". Usually, the branch name should be `username/small-title`
+  or better `username/issuenumber-small-title` where `issuenumber` is the number of
+  the github issue number that is tackled.
+- Push any needed commits to your branch.
+- When the branch is nearly ready for review, please open a pull request, and add the label `check-on-arm`
```
Comment (sguaschsbt) [inline CONTRIBUTING.md:35]: check-on-arm --> check-on-arm-64

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+- Do as many commits as necessary until all CI passes and all PR comments have been resolved.
```
Comment (sguaschsbt) [inline CONTRIBUTING.md]: CI --> CI checks

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  _During the process, optionnally, you may use `git rebase -i` to clean up your commit history. If you elect to do so,
+  please at the very least make sure that nobody else is working or has forked from your branch, otherwise the conflicts
+  it
+  will trigger are not worth it._
+- Finally, when all reviews are positive and all CI passes, you may merge your branch via the github webpage.
```
Comment (sguaschsbt) [inline CONTRIBUTING.md]: CI --> CI checks

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+To implement some changes, please follow these steps:
+
+- Create a "topic branch". Usually, the branch name should be `username/small-title`
+  or better `username/issuenumber-small-title` where `issuenumber` is the number of
+  the github issue number that is tackled.
+- Push any needed commits to your branch.
+- When the branch is nearly ready for review, please open a pull request, and add the label `check-on-arm`
+- Do as many commits as necessary until all CI passes and all PR comments have been resolved.
+  _During the process, optionnally, you may use `git rebase -i` to clean up your commit history. If you elect to do so,
```
Comment (sguaschsbt) [inline CONTRIBUTING.md]: optionnally --> optionally

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+### Versions and History
+
+* The project uses semantic versioning on the functions that are listed as `stable` in the documentation. A version has
+  the form `x.y.z`
+  * a patch release that increments `z` does not modify the stable API.
+  * a minor release that increments `y` adds a new feature to the stable API.
+  * In the unlikely case where we need to change or remove a feature, we will trigger a major release that
+    increments `x`.
+
+    _However, we will marked those features as deprecated at least 6 months before the major release._
```
Comment (sguaschsbt) [inline CONTRIBUTING.md]: marked --> mark

**Comment by sguaschsbt** [comment]
@ngama75 , @MGeorgie added some comments and suggestions for rewording. 
I understand the information for external collaborators will be added in another PR right?
Should we add something about enabling the WARNING_PARANOID flag when developing?

**Comment by ngama75** [comment]
That's a good idea, we should add:
* ensure that the source-code is formatted according to the clang-format style
* compiles with -DWARNING_PARANOID
* runs the unittests and integration tests locally
* has the flag check-on-arm64 on the Pull Request (Maintainer will set it after the review)

**Comment by sguaschsbt** [comment]
> That's a good idea, we should add:
> 
> * ensure that the source-code is formatted according to the clang-format style
> * compiles with -DWARNING_PARANOID
> * runs the unittests and integration tests locally
> * has the flag check-on-arm64 on the Pull Request (Maintainer will set it after the review)

I saw the changes! did you want to add something about unit tests at the end? I didn't see that

**Comment by ngama75** [comment]
No, I think the current page already covers it. We will revisit if we add long tests that are not covered by CI.

**Comment by sguaschsbt** [comment]
> No, I think the current page already covers it. We will revisit if we add long tests that are not covered by CI.

Sounds good :D

**Review by sguaschsbt — APPROVED** [review]
Looks good!

**Pull request merged**